src/main/java/fi/aalto/cs/apluscourses/utils/content/ZippedFile.java

```diff
@@ -0,0 +1,24 @@
+package fi.aalto.cs.apluscourses.utils.content;
+
+import fi.aalto.cs.apluscourses.utils.RemoteFileCache;
+import java.io.IOException;
+import java.nio.file.Path;
+import net.lingala.zip4j.ZipFile;
+import org.jetbrains.annotations.NotNull;
+
+public class ZippedFile implements Content {
+  private final String url;
+  private final String path;
+  private final RemoteFileCache remoteFileCache = RemoteFileCache.getInstance();
+
+  public ZippedFile(@NotNull String url, @NotNull String path) {
+    this.url = url;
+    this.path = path;
+  }
+
+  @Override
+  public void copyTo(@NotNull Path destinationPath) throws IOException {
+    new ZipFile(remoteFileCache.getCachedFile(url))
+        .extractFile(path, destinationPath.toString(), path.substring(path.lastIndexOf('/') + 1));
```

Comment:
Seemed like a perfect opportunity to use `Path`, but after some research it turns out ZIPs always use `/`, which `Path#toString` might not. I assume that's the reason for working with strings here

Reply:
Exactly.